crates/rpc/rpc-eth-api/src/helpers/spec.rs

```diff
@@ -8,21 +8,21 @@ use reth_errors::{RethError, RethResult};
 use reth_network_api::NetworkInfo;
 use reth_provider::{BlockNumReader, ChainSpecProvider, StageCheckpointReader};
 
-use super::EthSigner;
+use crate::{helpers::EthSigner, RpcNodeCore};
 
 /// `Eth` API trait.
 ///
 /// Defines core functionality of the `eth` API implementation.
 #[auto_impl::auto_impl(&, Arc)]
-pub trait EthApiSpec: Send + Sync {
-    /// Returns a handle for reading data from disk.
-    fn provider(
-        &self,
-    ) -> impl ChainSpecProvider<ChainSpec: EthereumHardforks> + BlockNumReader + StageCheckpointReader;
-
-    /// Returns a handle for reading network data summary.
-    fn network(&self) -> impl NetworkInfo;
-
+pub trait EthApiSpec:
+    RpcNodeCore<
+        Provider: ChainSpecProvider<ChainSpec: EthereumHardforks>
+                      + BlockNumReader
+                      + StageCheckpointReader,
+        Network: NetworkInfo,
+    > + Send
+    + Sync
+{
```

Comment:
cool, I like this a lot, this allows us to extend the `RpcNodeCore` internals as we see fit, which will be important to tackle #12023